app/views/assessments/_partial_feedback_js.html.erb

```diff
@@ -0,0 +1,65 @@
+<script>
+  function refreshPartial() {
+    $.ajax({
+        url: `getPartialFeedback?job_id=${<%= @job_id %>}`,
+        success: (res) => {
+          if (!res) return;
+          if (res.partial_feedback) {
+            $('#partial-feedback').html(`
+              <pre>${res.partial_feedback}</pre>
+              <div class="autograding-in-progress">
+                <pre>Autograding in progress</pre>
+              </div>`);
+            $('.feedback-status').html("In-Progress");
+            $('.feedback-status').removeClass("feedback-status__queued");
+            $('.feedback-status').addClass("feedback-status__inprogress");
+            $( "#result-body" ).load(window.location.href + " #result-body" );
+
+          } else if (res.queue_position != null) {
+            $('#partial-feedback').html(`
+              <div class="autograding-in-progress">
+                <pre>You are at position ${(res.queue_position) + 1} / ${(res.queue_length)} in the autograding queue.</pre>
+              </div>`);
+              $('#status-badge').html("Queued");
+              $('#status-badge').removeClass("feedback-status__inprogress");
+              $('#status-badge').addClass("feedback-status__queued");
+              $('#result-body').html(`
+              <h4>Submission queued...</h4>
+              <p>
+                Your submission is currently in the autograding queue, waiting to be graded.
+              </p>
+              <p>
+                Press the refresh button to get updates.
+              </p>`);
+          } else if (!res.is_assigned) {
+            /* if there's no partial feedback and no queue position,
+                refresh the page to see if full feedback is available */
+            location.reload();
+          }
+        },
+    });
+  }
+
+  const btn = $('#refresh-feedback')[0];
+
+  // Throttling Function
+  const throttleFunction=(func, delay)=>{
```

Comment:
We could use lodash's throttle function. https://lodash.com/docs/3.10.1#throttle 

We have already included lodash into our codebase as a dependency.  Refer to 
/Autolab/app/helpers/application_helper.rb and 
/Autolab/app/views/metrics/index.html.erb